planning.ipynb

```diff
@@ -594,6 +1104,249 @@
+       "    <span class=\"n\">effect_rem</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s1\">&#39;Have(Cake)&#39;</span><span class=\"p\">)]</span>\n",
+       "    <span class=\"n\">eat_cake</span> <span class=\"o\">=</span> <span class=\"n\">Action</span><span class=\"p\">(</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s1\">&#39;Eat(Cake)&#39;</span><span class=\"p\">),</span> <span class=\"p\">[</span><span class=\"n\">precond_pos</span><span class=\"p\">,</span> <span class=\"n\">precond_neg</span><span class=\"p\">],</span> <span class=\"p\">[</span><span class=\"n\">effect_add</span><span class=\"p\">,</span> <span class=\"n\">effect_rem</span><span class=\"p\">])</span>\n",
+       "\n",
+       "    <span class=\"c1\"># Bake Cake</span>\n",
+       "    <span class=\"n\">precond_pos</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
+       "    <span class=\"n\">precond_neg</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s1\">&#39;Have(Cake)&#39;</span><span class=\"p\">)]</span>\n",
+       "    <span class=\"n\">effect_add</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s1\">&#39;Have(Cake)&#39;</span><span class=\"p\">)]</span>\n",
+       "    <span class=\"n\">effect_rem</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
+       "    <span class=\"n\">bake_cake</span> <span class=\"o\">=</span> <span class=\"n\">Action</span><span class=\"p\">(</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s1\">&#39;Bake(Cake)&#39;</span><span class=\"p\">),</span> <span class=\"p\">[</span><span class=\"n\">precond_pos</span><span class=\"p\">,</span> <span class=\"n\">precond_neg</span><span class=\"p\">],</span> <span class=\"p\">[</span><span class=\"n\">effect_add</span><span class=\"p\">,</span> <span class=\"n\">effect_rem</span><span class=\"p\">])</span>\n",
+       "\n",
+       "    <span class=\"k\">return</span> <span class=\"n\">PDDL</span><span class=\"p\">(</span><span class=\"n\">init</span><span class=\"p\">,</span> <span class=\"p\">[</span><span class=\"n\">eat_cake</span><span class=\"p\">,</span> <span class=\"n\">bake_cake</span><span class=\"p\">],</span> <span class=\"n\">goal_test</span><span class=\"p\">)</span>\n",
+       "</pre></div>\n",
+       "</body>\n",
+       "</html>\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "psource(have_cake_and_eat_cake_too)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Have(x):** Declraes that we have **' x '**."
```

Comment:
Noticed this typo: 'Declraes' instead of 'Declares'.

Reply:
Thanks. Fixed that.